### PR TITLE
Enable deployments and replica sets in rsh

### PR DESCRIPTION
**Pull request by mfojtik**
Fixes: https://github.com/openshift/origin/issues/10756

### CODE REVIEW

**Comment by mfojtik** [comment]
@kargakis PTAL

**Comment by mfojtik** [comment]
[test]

**Comment by mfojtik** [comment]
@kargakis added test and consolidated messages with upstream

**Comment by mfojtik** [comment]
[test]

**Comment by mfojtik** [comment]
flake was: https://github.com/openshift/origin/issues/10228

File: test/cmd/rsh.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+# Cleanup cluster resources created by this test
+(
+  set +e
+  oc delete all,deployments --all
+  exit 0
+) &>/dev/null
+
+os::test::junit::declare_suite_start "cmd/rsh"
+os::cmd::expect_success 'oc run test-dc --image=openshift/origin --replicas=2 --command -- /bin/bash -c "sleep infinity"'
+os::cmd::expect_success 'kubectl run test-d --image=openshift/origin --replicas=2 --command -- /bin/bash -c "sleep infinity"'
```
Comment (kargakis) [inline test/cmd/rsh.sh]: Is the use of `kubectl` intentional?

**Comment by kargakis** [comment]
Just a question, LGTM

File: test/cmd/rsh.sh (hunk 1 above)
Comment (kargakis) [inline test/cmd/rsh.sh]: Can't you run `oc` with the deployment generator? This may break us the next time the default generator changes upstream.

Comment (mfojtik) [inline test/cmd/rsh.sh]: @kargakis on the other hand it is maybe good to be notified about such breakage, but sure I can change this to deployment generator.

Comment (kargakis) [inline test/cmd/rsh.sh]: Doesn't affect `oc` since we default to our dc generator.

**Comment by mfojtik** [comment]
@kargakis 
```
In suite "github.com/openshift/origin/test/cmd/rsh", test case "test/cmd/rsh.sh:16: executing 'oc rollout status dc/test-dc' expecting success and text 'rolled out'" failed:
=== BEGIN TEST CASE ===
test/cmd/rsh.sh:16: executing 'oc rollout status dc/test-dc' expecting success and text 'rolled out'
FAILURE after 3600.431s: test/cmd/rsh.sh:16: executing 'oc rollout status dc/test-dc' expecting success and text 'rolled out': the command returned the wrong error code; the output content test failed
Standard output from the command:
Waiting for rollout to finish: 0 out of 2 new replicas have been updated...

Standard error from the command:
error: timed out waiting for the condition
```

it seems like this PR will introduce new flakes when waiting for the replicas to be up. I tried to use the images already pulled on node...

**Comment by mfojtik** [comment]
[test]

@kargakis updated to use `oc`. I think I want this for 1.4 as it seems relatively small change for relatively big benefit :-) WDYT?

File: pkg/deploy/controller/deploymentconfig/controller.go [hunk 2]
```diff
@@ -351,24 +351,24 @@ func updateConditions(config deployapi.DeploymentConfig, newStatus *deployapi.De
...
 				deployutil.SetDeploymentCondition(newStatus, *condition)
 			}
 		case deployapi.DeploymentStatusFailed:
-			msg := fmt.Sprintf("Replication controller %q has failed progressing", latestRC.Name)
+			msg := fmt.Sprintf("replication controller %q has failed progressing", latestRC.Name)
 			condition := deployutil.NewDeploymentCondition(deployapi.DeploymentProgressing, kapi.ConditionFalse, deployutil.TimedOutReason, msg)
 			deployutil.SetDeploymentCondition(newStatus, *condition)
 		case deployapi.DeploymentStatusComplete:
-			msg := fmt.Sprintf("Replication controller %q has completed progressing", latestRC.Name)
+			msg := fmt.Sprintf("replication controller %q successfully rolled out", latestRC.Name)
```
Comment (kargakis) [inline pkg/deploy/controller/deploymentconfig/controller.go]: With this change:
```sh
$ oc rollout status dc/foo
...
replication controller "foo-1" successfully rolled out

$ oc rollout status deploy/foo
...
deployment "foo" successfully rolled out
```
We should be consistent.

Comment (mfojtik) [inline pkg/deploy/controller/deploymentconfig/controller.go]: \o/

**Comment by mfojtik** [comment]
@kargakis ping

**Comment by kargakis** [comment]
LGTM [merge]

File: pkg/cmd/util/clientcmd/factory.go [hunk 3]
```diff
@@ -976,6 +976,42 @@ func (f *Factory) PodForResource(resource string, timeout time.Duration) (string
...
+		selector, err := unversioned.LabelSelectorAsSelector(rs.Spec.Selector)
+		if err != nil {
+			return "", err
+		}
+		pod, _, err := cmdutil.GetFirstPod(kc, namespace, selector, timeout, sortBy)
+		if err != nil {
+			return "", err
+		}
+		return pod.Name, nil
 	case extensions.Resource("jobs"):
```
Comment (kargakis) [inline pkg/cmd/util/clientcmd/factory.go]: @soltysh should we drop this part of code in 1.5? Also we may want to add support for cronjobs?

Comment (soltysh) [inline pkg/cmd/util/clientcmd/factory.go]: extensions are still in 1.5, but cronjobs should be added here.

**Comment by soltysh** [comment]
Flake https://github.com/openshift/origin/issues/11452

[merge]

**Pull request merged**